api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/data/quality/Ec20Queries.java

```diff
@@ -0,0 +1,47 @@
+package org.openmrs.module.eptsreports.reporting.library.queries.data.quality;
+
+public class Ec20Queries {
+
+  /**
+   * EC20 Patients that are not enrolled in TARV but has a consultation or drugs pick up recorded in
+   * the system
+   *
+   * @param programId
+   * @param arvPediatriaSeguimentoEncounterType
+   * @param adultoSeguimentoEncounterType
+   * @param arvPediatriaSeguimentoEncounterType
+   * @return
+   */
+  public static String getEc20CombinedQuery(
+      int programId,
+      int arvPediatriaSeguimentoEncounterType,
+      int adultoSeguimentoEncounterType,
+      int arvPharmaciaEncounterType) {
+    String query =
+        "SELECT patient_id, NID, Name, birthdate, Estimated_dob, Sex, First_entry_date, Last_updated,"
+            + " CASE WHEN encounter_type=6 THEN encounter_date WHEN encounter_type=9 THEN encounter_date END AS first_consultation_date,"
+            + " CASE WHEN encounter_type=18 THEN encounter_date END AS first_drug_pickup_date FROM"
+            + " (SELECT pa.patient_id, pi.identifier AS NID, CONCAT(pn.given_name, ' ', pn.family_name ) AS Name, DATE_FORMAT(pe.birthdate, '%d-%m-%Y') AS birthdate,"
+            + " IF(pe.birthdate_estimated = 1, 'Yes','No') AS Estimated_dob, pe.gender AS Sex, DATE_FORMAT(pa.date_created, '%d-%m-%Y %H:%i:%s') AS First_entry_date,"
+            + " DATE_FORMAT(pa.date_changed, '%d-%m-%Y %H:%i:%s') AS Last_updated,DATE_FORMAT(pg.date_enrolled, '%d-%m-%Y %H:%i:%s') AS date_enrolled,"
+            + " DATE_FORMAT(e.encounter_datetime, '%d-%m-%Y %H:%i:%s') AS encounter_date, DATE_FORMAT(e.date_created, '%d-%m-%Y %H:%i:%s') AS encounter_date_created,"
+            + " e.encounter_type FROM patient pa"
+            + " INNER JOIN patient_identifier pi ON pa.patient_id=pi.patient_id"
+            + " INNER JOIN person pe ON pa.patient_id=pe.person_id"
+            + " INNER JOIN person_name pn ON pa.patient_id=pn.person_id"
+            + " INNER JOIN patient_program pg ON pa.patient_id=pg.patient_id"
+            + " INNER JOIN encounter e ON pa.patient_id=e.patient_id"
+            + " WHERE pg.program_id!="
+            + programId
+            + " AND e.voided=0  AND e.encounter_type IN ("
+            + arvPediatriaSeguimentoEncounterType
+            + ","
+            + adultoSeguimentoEncounterType
+            + ","
```

Comment:
I have seen seen requirement **The system will identify all patients who have clinical appointment before 1985** in your query

Reply:
@ningosi this is the requirement for EC20: **Patients that are not enrolled in TARV but has a consultation or drugs pick up recorded in the system**